retriever/engines/mysql.py

```diff
@@ -39,6 +41,21 @@ class engine(Engine):
                       "{db}.{table}"),
                      ]
 
+    def check_credentials(self):
+        self.get_input()
+        self.sql_lookup = {"user": self.opts["user"], "password": self.opts["password"], "host": self.opts["host"], "port": self.opts["port"]}
```

Comment:
Yes, I feel the same way. I always try to work with local variables. For this case I think we can just update the` self.opts`.

Lets include a doc string for def check_credentials()